src/django_secret_sharing/mixins.py

```diff
@@ -62,3 +67,27 @@ def get_secret_url(self, secret):
         return self.request.build_absolute_uri(
             reverse("django_secret_sharing:retrieve", kwargs={"hash": secret})
         )
+
+
+class ViewSecretMixin(SecretsMixin):
+    def get_view_context(self, hash):
+        context = {}
+        try:
+            context["secret"] = self.decrypt_secret(hash)
+        except DecryptError:
+            raise Http404()
+        return context
+
+
+class RetrieveSecretMixin(SecretsMixin):
+    def get_retrieve_context(self, hash):
+        context = {}
+        if not self.validate_secret(hash):
+            raise Http404()
+        context["url_part"] = hash
+        return context
+
+
+class CreateSecretMixin(SecretsMixin):
+    def get_create_context(self):
+        return {"form": CreateSecretForm()}
```

Comment:
how do you handlle a submitted form, the form should be passed to this method instaed (as you probably need CreateSecretForm(request.data) as well as CreateSecretForm())